tests/unit/Beer.spec.js

```diff
@@ -0,0 +1,70 @@
+import { shallowMount } from '@vue/test-utils';
+import Beer from '@/components/Beer';
+
+// Beer object from api to test against.
+const beer = {
+  id: 192,
+  name: 'Punk IPA 2007 - 2010',
+  tagline: 'Post Modern Classic. Spiky. Tropical. Hoppy.',
+  description:
+    "Our flagship beer that kick started the craft beer revolution. This is James and Martin's original take on an American IPA, subverted with punchy New Zealand hops. Layered with new world hops to create an all-out riot of grapefruit, pineapple and lychee before a spiky, mouth-puckering bitter finish.",
+  image_url: 'https://images.punkapi.com/v2/192.png',
+  abv: 6,
+  food_pairing: [
+    'Spicy carne asada with a pico de gallo sauce',
+    'Shredded chicken tacos with a mango chilli lime salsa',
+    'Cheesecake with a passion fruit swirl sauce',
+    'We only want to show three food pairings, our computed property should remove it',
+  ],
+};
+
+// https://lmiller1990.github.io/vue-testing-handbook/components-with-props.html#refactor-with-a-factory-function
+const factory = (propsData) => {
+  return shallowMount(Beer, {
+    propsData: {
+      beer,
+      ...propsData,
+    },
+  });
+};
+
+describe('Beer', () => {
+  it('renders without errors', () => {
+    const wrapper = factory();
+    expect(wrapper.isVueInstance()).toBeTruthy();
+  });
+
+  it('renders beer image attributes', () => {
+    const wrapper = factory();
+    expect(wrapper.find('.beer__img').attributes('src')).toBe(
+      'https://images.punkapi.com/v2/192.png',
+    );
+    expect(wrapper.find('.beer__img').attributes('alt')).toBe(beer.name);
```

Comment:
Now you have a few test checking for everything to be displayed, what if you refactored it, with different styles and changed the classes, but the same data are displayed?

To be fair, your BEM naming is pretty good and unlikely to break those tests, but the point is relying on styling classes makes your test likely to break in the future for no good reason.

There is a few options here to prevent it: you can use a class called test-{name}. But the most used method which is pretty standard is to use a data attribute like data-test=“beer-image” and then you can do [data-test=“beer-image”].

Reply:
This makes perfect sense adding data-test attributes instead of relying on classes to limit the chance of tests breaking.

I would like to have them removed from build, I found this and it works 👍 
https://forum.vuejs.org/t/how-to-remove-attributes-from-tags-inside-vue-components/24138/9